core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java

```diff
@@ -34,172 +24,169 @@
+
+    public NativeImpl load() {
+      try {
+        return new JnrNativeImpl();
+      } catch (Throwable t) {
+        LOG.info(
+            "Unable to load JNR native implementation. This could be normal if JNR is excluded from the classpath",
+            t);
+        return new EmptyNativeImpl();
+      }
+    }
+  }
+
+  /* Copied from equivalent op in jnr.ffi.Platform.  We have to have this here as it has to be defined
+   * before its (multiple) uses in determineCpu() */
+  private static final Locale LOCALE = Locale.ENGLISH;
+
+  private static final NativeImpl IMPL = new ImplLoader().load();
+
+  @SuppressWarnings("VariableNameSameAsType")
+  private static final Cpu CPU = determineCpu();
+
   /** Whether {@link Native#currentTimeMicros()} is available on this system. */
   public static boolean isCurrentTimeMicrosAvailable() {
-    try {
-      return LibCLoader.GET_TIME_OF_DAY_AVAILABLE;
-    } catch (NoClassDefFoundError e) {
-      return false;
-    }
+    return IMPL.gettimeofdayAvailable();
   }
 
   /**
    * The current time in microseconds, as returned by libc.gettimeofday(); can only be used if
    * {@link #isCurrentTimeMicrosAvailable()} is true.
    */
   public static long currentTimeMicros() {
-    if (!isCurrentTimeMicrosAvailable()) {
-      throw new IllegalStateException(
-          "Native call not available. "
-              + "Check isCurrentTimeMicrosAvailable() before calling this method.");
-    }
-    LibCLoader.Timeval tv = new LibCLoader.Timeval(LibCLoader.LIB_C_RUNTIME);
-    int res = LibCLoader.LIB_C.gettimeofday(tv, null);
-    if (res != 0) {
-      throw new IllegalStateException("Call to libc.gettimeofday() failed with result " + res);
-    }
-    return tv.tv_sec.get() * 1000000 + tv.tv_usec.get();
+    return IMPL.gettimeofday();
   }
 
   public static boolean isGetProcessIdAvailable() {
-    try {
-      return PosixLoader.GET_PID_AVAILABLE;
-    } catch (NoClassDefFoundError e) {
-      return false;
-    }
+    return IMPL.getpidAvailable();
   }
 
   public static int getProcessId() {
-    if (!isGetProcessIdAvailable()) {
-      throw new IllegalStateException(
-          "Native call not available. "
-              + "Check isGetProcessIdAvailable() before calling this method.");
-    }
-    return PosixLoader.POSIX.getpid();
+    return IMPL.getpid();
   }
 
   /**
-   * Returns {@code true} if JNR {@link Platform} class is loaded, and {@code false} otherwise.
+   * Returns the current processor architecture the JVM is running on. This value should match up to
+   * what's returned by jnr-ffi's Platform.getCPU() method.
    *
-   * @return {@code true} if JNR {@link Platform} class is loaded.
+   * @return the current processor architecture.
    */
-  public static boolean isPlatformAvailable() {
-    try {
-      return PlatformLoader.PLATFORM != null;
-    } catch (NoClassDefFoundError e) {
-      return false;
-    }
+  public static String getCpu() {
```

Comment:
Since this is class is becoming a shell that delegates all of its operations to a `NativeImpl` instance, wouldn't it be a better design to also include `getCpu()` in `NativeImpl`?

Reply:
Or better yet: since determining the CPU does not depend on any native call, just on system properties. Shouldn't we then move all CPU stuff to its own class?

Reply:
Since we're preserving Native as the main entry point into this functionality I'd argue for keeping the functionality here.  Given it's nature this functionality won't change in the various NativeImpl/Libc implementations so I'd argue against pushing it down to that level.

Your point about moving it to it's own class is interesting, though.  I'd be fine moving this functionality to a discrete class which is then used by Native.getCpu() to implement the necessary functionality.  I'll take a quick stab at that and see what it looks like.

Reply:
Done; all this logic has now been moved to the CpuInfo class